### DISCUSSION TITLE
Credentials - BasicAuth

### DISCUSSION THREAD

**Discussion by dawidwekwejt** [serilog-contrib/serilog-sinks-grafana-loki#143] [Q&A]
How can I defined basic auth? Below there is my config:

```
"Serilog": {
    "Using": [
      "Serilog.Sinks.Grafana.Loki"
    ],
    "MinimumLevel": {
      "Default": "Debug"
    },
    "WriteTo": [
      {
        "Name": "********",
        "Args": {
          "uri": "********",
          "credentials": {
            "user": "********",
            "password": "********"
          },
          "labels": [
            {
              "key": "app",
              "value": "********"
            },
            {
              "key": "environment",
              "value": "development"
            }
          ],
          "propertiesAsLabels": [
            "app"
          ]
        }
      }
    ]
  }
```

**Comment by mishamyte** [comment]
Hello @dawidwekwejt,

You should change property name `user` to `login`, because it is defined in [LokiCredentials](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/blob/master/src/Serilog.Sinks.Grafana.Loki/LokiCredentials.cs), to which you configuration section would be binded

**Reply by dawidwekwejt** [reply]
I still can't sent messages to Loki. With logger configured and built with code everythink is ok. Is my configuration structute right? I can't find any sample with authentication...

**Reply by mishamyte** [reply]
Uhm, yep. Planning to cover more cases in the doc.

For me it seems right, but I will need to check when I will return to PC.

For now, I could suggest 2 variants:
1. Configure from code and ensure that security header is passed and works 
2. Enable `SelfLog` and check it for errors

**Reply by mishamyte** [reply]
Hi, as promised I'm checking the case.

So I used the next configuration for `nginx` before Loki:
```
  nginx:
    image: laurentbel/nginx-basic-auth
    ports:
      - "80:80"
    depends_on:
      - loki
    environment:
      - FORWARD_HOST=loki
      - FORWARD_PORT=3100
      - BASIC_USERNAME=foo
      - BASIC_PASSWORD=bar
    networks:
      - loki
 ```
 
 Rest of `docker-compose` from this repo is untouched.
 
 And used web-app from `samples` of this repo, extended config. So my config is next:
 ```json
 {
  "Serilog": {
    "Using": [
      "Serilog.Sinks.Console",
      "Serilog.Sinks.Grafana.Loki"
    ],
    "MinimumLevel": {
      "Default": "Debug",
      "Override": {
        "Microsoft": "Warning",
        "System": "Warning"
      }
    },
    "Enrich": [
      "WithThreadId"
    ],
    "WriteTo": [
      {
        "Name": "Console",
        "Args": {
          "outputTemplate": "{Timestamp:dd-MM-yyyy HH:mm:ss} [{Level:u3}] [{ThreadId}] {Message}{NewLine}{Exception}"
        }
      },
      {
        "Name": "GrafanaLoki",
        "Args": {
          "uri": "http://localhost:80",
          "credentials": {
            "login": "foo",
            "password": "bar"
          },
          "labels": [
            {
              "key": "app",
              "value": "web_app"
            }
          ],
          "propertiesAsLabels": [
            "app"
          ]
        }
      }
    ]
  }
}
 ```
 
 And all works like a charm. Also I checked that without `credentials` section `nginx` returns 401 and events are not delivered to Loki so far.
 
 Also I could see use masked the `Name` section of your config. It **MUST** be `GrafanaLoki` for binder to work.
 
 Could you use config above as an example?
 
 Also, you will have a possibility to see the errors in your app's console, if you will add the next line to your service's code
 ```csharp
 SelfLog.Enable(Console.Error);
 ```
